hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1042,22 +1046,27 @@ public static void aggregateColumnStats(IndexedRecord record, Schema schema,
 
     schema.getFields().forEach(field -> {
       Map<String, Object> columnStats = columnToStats.getOrDefault(field.name(), new HashMap<>());
-      final String fieldVal = getNestedFieldValAsString((GenericRecord) record, field.name(), true, consistentLogicalTimestampEnabled);
+      final Object fieldVal = getNestedFieldVal((GenericRecord) record, field.name(), true, consistentLogicalTimestampEnabled);
```

Comment:
Few notes: 

1. `getNestedFieldVal` is too heavy-weight of an operation, which we don't actually need here (we don't need to parse the dot-path, etc). Instead we can directly access the field value and perform logical type conversion by invoking `convertValueForSpecificDataTypes` directly.
2. `fieldVal` we're getting should be of type `Comparable<?>`, otherwise we set it to null (since there's no plausible order we can impose on these

Reply:
Also, since we rely on Parquet for collecting stats for base files we need to make sure that this sequence is _compatible_ with it.

As such, we can't use comparator for Avro records and assume it would be identical to that one of Parquet (Thrift) as it's very likely to be not: since binary representation of this encodings is different we can't really compare composite records like that (arrays, maps are all treated like binary in Parquet, see ref at the bottom).

As such my proposal is to limit min/max stats only for primitive types.

Ref: Take a look at [PrimitiveComparator](https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveComparator.java) implementation